cypress/integration/chi-vue/data-table.spec.js

```diff
@@ -1,10 +1,888 @@
+      cy.get(`[data-cy='data-table-accordion'] .${DATA_TABLE_CLASSES.BODY}`)
+        .find(`.${DATA_TABLE_CLASSES.EXPANDABLE} button`)
+        .eq(0)
+        .as('expandableButton')
+        .click()
+        .then(() => {
+          hasClassAssertion('@rows', '-expanded');
+          cy.get('@childRowContainer').should(
+            'not.have.css',
+            'display',
+            'none'
+          );
+        });
+      cy.get('@expandableButton')
+        .click()
+        .then(() => {
+          cy.get('@rows').should('not.have.class', '-expanded');
+        });
+    });
+  });
+
+  describe('sorting', () => {
+    describe('base', () => {
+      beforeEach(() => {
+        cy.get(`[data-cy='data-table-sorting'] .${DATA_TABLE_CLASSES.BODY}`)
+          .find(`.${DATA_TABLE_CLASSES.ROW}`)
+          .eq(0)
+          .as('firstRow');
+      });
+
+      it(`should have class .${DATA_TABLE_CLASSES.SORTING}`, () => {
+        hasClassAssertion(
+          `[data-cy='data-table-sorting'] .${DATA_TABLE_CLASSES.CELL}`,
+          DATA_TABLE_CLASSES.SORTING
+        );
+      });
+
+      it.skip('should trigger the appropriate event', () => {
+        // TODO: add test once workaround is found.
+        const spy = cy.spy();
+      });
+
+      it(`should sort by name with the correct icons`, () => {
+        cy.get('@firstRow').should('contain', 'Name 1');
+        cy.get(`[data-cy='data-table-sorting'] .${DATA_TABLE_CLASSES.ROW}`)
+          .eq(0)
+          .find(`.${DATA_TABLE_CLASSES.CELL}`)
+          .eq(0)
+          .as('firstCell')
+          .find('.chi-icon')
+          .as('sortIcon');
+        hasClassAssertion('@sortIcon', 'icon-arrow-sort');
+        cy.get('@firstCell')
+          .click()
+          .then(() => {
+            hasClassAssertion('@sortIcon', 'icon-arrow-up');
+            cy.get('@firstRow').should('contain', 'Name 1');
+            cy.get('@sortIcon').should('have.css', 'transform', 'none');
+          });
+        cy.get('@firstCell')
+          .click()
+          .then(() => {
+            hasClassAssertion('@sortIcon', 'icon-arrow-up');
+            cy.get('@sortIcon')
+              .should('have.attr', 'style')
+              .should('contain', 'transform: rotate(180deg);');
+            cy.get('@firstRow').should('contain', 'Surname 2');
+          });
+        cy.get('@firstCell')
+          .click()
+          .then(() => {
+            cy.get('@firstRow').should('contain', 'Name 1');
+            hasClassAssertion('@sortIcon', 'icon-arrow-sort');
+          });
+      });
+
+      it(`should sort by status using a custom template`, () => {
+        cy.get('@firstRow').should('contain', 'Name 1');
+        cy.get(`[data-cy='data-table-sorting'] .${DATA_TABLE_CLASSES.ROW}`)
+          .eq(0)
+          .find(`.${DATA_TABLE_CLASSES.CELL}`)
+          .eq(1)
+          .as('idCell')
+          .click()
+          .then(() => {
+            cy.get('@firstRow').should('contain', 'active');
```

Comment:
You are checking the content of the cell of the first row only. Please, check the content of all the visible rows
Example:
```
active
active
active
inactive
inactive
```
Click sorting again and
```
inactive
inactive
active
active
active
```